### PR TITLE
liblrs proposal

### PR DESCRIPTION
**Pull request by loic-hamelin**
Questionnaire for liblrs project.

### CODE REVIEW

File: project_proposals/liblrs.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+##  What is the project's name?
+
+libLRS
+
+It stands for _library for linear referencing systems_.
+
+## Describe the project. What does the project do, why is it valuable, where does it come from?
+
+In the railway industry, most of the objects are located by a linear referencing system, a position along a track (i.e. : track 1, kilometer 3+532 meters).
```
Comment (Tristramg) [inline project_proposals/liblrs.md:11]:
```suggestion
In the railway industry, most of the objects are located by a linear referencing system, a position along a track (i.e.: track 1, kilometer 3+532 meters).
```

File: project_proposals/liblrs.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+This library has the ambition to be a cornerstone of any tool concerning rail networks. As such, being included in the OpenRail Association will allow the library to be considered as a neutral solution and give more visibility to become the reference library for LRSs.
+
+## Are there competing products or projects? If there are please explain how the proposed projects differentiates.
+
+GIS software (like ARCGis and QGIS) have plugins to manipulate those data. However they cannot be used with existing software, as OSRD for example.
+
+## What standards does the project implement or rely on? How are they related to other existing standards?
+
+
```
Comment (Tristramg) [inline project_proposals/liblrs.md:119]:
```suggestion
The data representation and the vocabulary is based on the article _A generic Data Model for Linear Referencing Systems_ published in the National Cooperative Highway Research Program, Research Result Digest (number 218, September 1997) and the [ISO 19148](https://www.iso.org/standard/75150.html) norm.
```

File: project_proposals/liblrs.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+## What is the tech stack of the project? Name the major programming languages and frameworks which are used.
+
+The core is developped in rust.
+
+Bindings are generated for python and javascript. Bindings for other languages will be added if asked.
+
+## What is the project's plan for growing in maturity if accepted within the OpenRail Association?
+
+* Validation of the proposed data model by other infrastructure managers
+* Code reviews from other users : optimization and reliability of the code
```
Comment (Tristramg) [inline project_proposals/liblrs.md:130]:
```suggestion
Code reviews from other users: optimization and reliability of the code
```

File: project_proposals/liblrs.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+It stands for _library for linear referencing systems_.
+
+## Describe the project. What does the project do, why is it valuable, where does it come from?
+
+In the railway industry, most of the objects are located by a linear referencing system, a position along a track (i.e. : track 1, kilometer 3+532 meters).
+
+As railways networks may be more than a century old, there are many exceptions, local specific rules…
+
+The goal of the libray is to have a flexible, high performance and easy to integrate linear referencing systems library that can be used in any system to manipulate LRSs:
```
Comment (SergeCroise) [inline project_proposals/liblrs.md:15]:
```suggestion
The goal of the library is to have a flexible, high performance and easy to integrate linear referencing systems library that can be used in any system to manipulate LRSs:
```

File: project_proposals/liblrs.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+* projecting from one linear referencing system to an other
```
Comment (SergeCroise) [inline project_proposals/liblrs.md:16]:
```suggestion
* projecting from one linear referencing system to another
```

File: project_proposals/liblrs.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+## Which exact repositories do you intend to  transfer to the GitHub organization of the OpenRail Association?
+
+https://github.com/osrd-project/liblrs/
+
+## What is the project's main license?
+
+MIT https://github.com/osrd-project/liblrs/blob/main/LICENSE.md
+
+## What other licenses does the project use, e.g. for included 3rd party code?
```
Comment (SergeCroise) [inline project_proposals/liblrs.md:43]:
```suggestion
## What other licenses does the project use, e.g., for included 3rd-party code?
```

File: project_proposals/liblrs.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+## How is it decided if and when a pull request is merged?
+
+The review of a maintainer is required.
+
+## How can someone become a committer or a maintainer to/of the project?
+
+Anybody can suggest a pull request. If it is accepted after the review, one becomes a commiter.
+
+Maintainers are coopted by the other maintainers. Candidates must be an active commiter.
```
Comment (SergeCroise) [inline project_proposals/liblrs.md:88]:
```suggestion
Maintainers are coopted by the other maintainers. Each candidate must be an active committer.
```

File: project_proposals/liblrs.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+As the team is small, all decisions are made collaboratively. We meet each week to share progress and decide on next steps. We are open to include any interested person into the decision-making team.
+
+## How is it decided if and when a pull request is merged?
+
+The review of a maintainer is required.
+
+## How can someone become a committer or a maintainer to/of the project?
+
+Anybody can suggest a pull request. If it is accepted after the review, one becomes a commiter.
```
Comment (SergeCroise) [inline project_proposals/liblrs.md:86]:
```suggestion
Anybody can suggest a pull request. If it is accepted after the review, one becomes a committer.
```

File: project_proposals/liblrs.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+GIS software (like ARCGis and QGIS) have plugins to manipulate those data. However they cannot be used with existing software, as OSRD for example.
+
+## What standards does the project implement or rely on? How are they related to other existing standards?
+
+
+
+## What is the tech stack of the project? Name the major programming languages and frameworks which are used.
+
+The core is developped in rust.
```
Comment (SergeCroise) [inline project_proposals/liblrs.md:123]:
```suggestion
The core is developed in rust.
```
according to:
https://www.chicagomanualofstyle.org/search.html?clause=developed

**Review by cornelius — APPROVED** [review]
This sounds like a very solid proposal. It's a well-defined project with an active group of maintainers and a clear governance structure. It's relatively small, but that's good for a building block other projects can use.

There are a few changes in documentation needed to meet the standards of our minimal viable governance templates (https://github.com/OpenRailAssociation/technical-committee/tree/main/project-templates), but I don't see any indication that this would be an issue. We can also sort this out when the repo is becoming part of the OpenRail Assocation organization.

**Comment by cornelius** [comment]
The Technical Committee has decided to accept the project on 2024-07-09.

**Comment by cornelius** [comment]
Accepted by vote of board on 2024-07-12.

**Pull request merged**